### PR TITLE
Better docs

### PR DESCRIPTION
**Pull request by shacharoo**
Solves #44.

@snudler6  - You can squash these when you merge.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+This is bad for several reasons:
+
+* **Your test takes 5 seconds to run**. That's a no-no.
+* ``time.sleep()`` promises that the process will sleep ``x`` seconds
+  **at most**. This test might fail randomly, depending on how sensitive your
+  state machine is.
+
+There's nothing worse than a heisenbuild (well, perhaps a **SLOW** heisenbuild).
+Here's the **better** way to do this using ``time-travel``:
```
Comment (snudler6) [inline README.rst]: A better

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+.. code-block:: python
+
+   def test_state_timeout():
+       with TimeTravel() as tt:
+           sm.handle_event(event=...)
+           tt.clock.time += 5
+           sm.handle_event(event=...)
+           assert sm.state == TIMEOUT
+
+When the ``handle_event`` method is called, it will probably check the time
```
Comment (snudler6) [inline README.rst]: Check the time

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+using the ``time`` or ``datetime`` module. These modules are mocked by
```
Comment (snudler6) [inline README.rst]: Using one of time or datetime

Comment (snudler6) [reply README.rst]: This modules are patched?

File: docs/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (snudler6) [inline docs/Makefile]: ?

File: docs/tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+   This is because Windows does not support any lower values. Sorry UNIX users!
+
+Performance Issues
+^^^^^^^^^^^^^^^^^^
+
+The way the context manager works is that it changes references to patched
+objects in loaded modules. By default ``time-travel`` searches through
+**every loaded module** in ``sys.modules``. **This takes around 2 seconds**.
+
+To minimize search time, ``time_travel.TimeTravel`` gets a keyword argument
```
Comment (snudler6) [inline docs/tutorial.rst]: Make this also bold so no one will stop read after the previous line :)
Also can make it funny? Something like "alas - we have the solution for you"

File: docs/implementation.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+The Event Pool
+--------------
+
+The event pool keeps a set of events for different file descriptors, in
+different timestamps. The pool's job is to keep those events and to retrieve
+them for different patchers.
+
+.. todo::
+
+   Document the patcher's interface with the pool.
```
Comment (snudler6) [inline docs/implementation.rst]: Now or issue?

File: README.rst (hunk 2 above)
Comment (shacharoo) [reply README.rst]: ?

File: README.rst (hunk 1 above)
Comment (shacharoo) [reply README.rst]: K

File: README.rst (hunk 3 above)
Comment (shacharoo) [reply README.rst]: K K

File: docs/Makefile (hunk 4 above)
Comment (shacharoo) [reply docs/Makefile]: Auto-generated by sphinx.

File: docs/implementation.rst (hunk 6 above)
Comment (shacharoo) [reply docs/implementation.rst]: Will do.

File: docs/tutorial.rst (hunk 5 above)
Comment (shacharoo) [reply docs/tutorial.rst]: KK

File: docs/implementation.rst (hunk 6 above)
Comment (shacharoo) [reply docs/implementation.rst]: #50 #51

**Pull request merged**